app/services/clients/products_client.js

```diff
@@ -91,9 +96,29 @@ function getProductsByGatewayAccountId (gatewayAccountId) {
   }).then(products => products.map(product => new Product(product)))
 }
 
+/**
+ * @param {String} gatewayAccountId: the external id of the product you wish to retrieve
+ * @returns {Promise<Product>}
+ */
+function updateServiceNameOfProductsByGatewayAccountId (gatewayAccountId, serviceName) {
```

Comment:
I still don' think products-ui ever needs this operation at all. any service name updates must come from selfservice only.